File: solutions/bytelandian_gold_coins/bytelandian_gold_coins.java
import java.util.*;

class bytelandian_gold_coins
{
    static HashMap<Integer,Long>c=new HashMap<>();
    public static long process(int n) 
    {
        if(n==0)
        {
            return n;
        }
        if(c.containsKey(n))
        {
            return c.get(n);
        }
        else
        {
            long ans = Math.max(n, process(n/2) + process(n/3) + process(n/4));
            c.put(n, ans);
            return ans;
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("type input");
        while(sc.hasNext()) {
			int n = sc.nextInt();
			System.out.println(process(n));
		}
        sc.close();        
    }
}

File: solutions/bytelandian_gold_coins/bytelandian_gold_coins.py
dic = {0:0}
def process(n):
    if n in dic:
        return dic[n]
    else:
        dic[n] = max(n, process(n//2) + process(n//3) + process(n//4))
        return dic[n]
while True:
    try:
        n = int(input())
        print(process(n))
    except expression as identifier:
        break
    
    